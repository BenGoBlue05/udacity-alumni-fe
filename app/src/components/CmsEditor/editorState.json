 {
  "entityMap": {
    "0": {
      "type": "LINK",
      "mutability": "MUTABLE",
      "data": {
        "href": "https://www.wired.com/author/scott-dadich/",
        "target": "_blank",
        "url": "https://www.wired.com/author/scott-dadich/"
      }
    }
  },
  "blocks": [
    {
      "key": "6vmpu",
      "text": "IT’S HARD TO think of a single technology that will shape our world more in the next 50 years than artificial intelligence. As machine learning enables our computers to teach themselves, a wealth of breakthroughs emerge, ranging from medical diagnostics to cars that drive themselves. A whole lot of worry emerges as well. Who controls this technology? Will it take over our jobs? Is it dangerous? President Obama was eager to address these concerns. The person he wanted to talk to most about them? Entrepreneur and MIT Media Lab director Joi Ito. So I sat down with them in the White House to sort through the hope, the hype, and the fear around AI. That and maybe just one quick question about Star Trek. —SCOTT DADICH",
      "type": "unstyled",
      "depth": 0,
      "inlineStyleRanges": [
        {
          "offset": 80,
          "length": 14,
          "style": "BOLD"
        },
        {
          "offset": 697,
          "length": 9,
          "style": "ITALIC"
        }
      ],
      "entityRanges": [
        {
          "offset": 709,
          "length": 12,
          "key": 0
        }
      ],
      "data": {}
    },
    {
      "key": "18f3q",
      "text": "",
      "type": "unstyled",
      "depth": 0,
      "inlineStyleRanges": [],
      "entityRanges": [],
      "data": {}
    },
    {
      "key": "69rel",
      "text": "",
      "type": "unstyled",
      "depth": 0,
      "inlineStyleRanges": [],
      "entityRanges": [],
      "data": {}
    },
    {
      "key": "8ejl3",
      "text": "",
      "type": "atomic",
      "depth": 0,
      "inlineStyleRanges": [],
      "entityRanges": [],
      "data": {
        "src": "https://www.wired.com/wp-content/uploads/2016/10/ito1-1.jpg",
        "type": "image",
        "display": "medium"
      }
    },
    {
      "key": "77qng",
      "text": "",
      "type": "unstyled",
      "depth": 0,
      "inlineStyleRanges": [],
      "entityRanges": [],
      "data": {}
    }
  ]
}
