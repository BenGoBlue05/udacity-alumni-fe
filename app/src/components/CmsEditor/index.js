import React, { PropTypes, Component } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import { MegadraftEditor, editorStateFromRaw } from 'megadraft';
import { CmsToolbar } from 'components';

class CmsEditor extends Component {
  render() {
    const {
      editorState,
      editorTitle,
      isValid,
      onChangeTitle,
      onSubmit,
      onChangeContent,
      onTapToPreview,
    } = this.props;
    const stateForEditor = editorState ? editorState : editorStateFromRaw(null);
    return (
      <div className={styles.container}>
        <div className={styles.inputContainer}>
          <input
            className={styles.input}
            ref="titleInput"
            type="titleInput"
            value={editorTitle}
            onChange={onChangeTitle}
            placeholder="Title"
          />
        </div>
        <div className={styles.cmsInput}>
          <MegadraftEditor
            editorState={stateForEditor}
            onChange={(state) => onChangeContent(state)}
            placeholder="Tell a story ..."
          />
        </div>
        <CmsToolbar
          canSubmit={isValid}
          onSave={onSubmit}
          onPreview={onTapToPreview}
        />
      </div>
    );
  }
}

CmsEditor.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onTapToPreview: PropTypes.func.isRequired,
  onChangeTitle: PropTypes.func.isRequired,
  onChangeContent: PropTypes.func.isRequired,
  editorTitle: PropTypes.string,
  editorState: PropTypes.object,
  isValid: PropTypes.bool.isRequired,
};

export default cssModules(CmsEditor, styles);
