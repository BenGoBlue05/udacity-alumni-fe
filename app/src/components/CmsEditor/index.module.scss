.container {
  padding: 3% 10%;
}

.button {
  display: block !important;
  margin: 30px auto !important;
}

.inputContainer {
  input,
  input::-webkit-input-placeholder {
    font-size:  2.5rem;
  }
}

.input {
  border: 0 !important;
  box-shadow: none !important;
}
