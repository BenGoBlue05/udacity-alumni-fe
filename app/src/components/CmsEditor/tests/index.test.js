import CmsEditor from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<CmsEditor />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <CmsEditor
        onSubmit={e => e}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
