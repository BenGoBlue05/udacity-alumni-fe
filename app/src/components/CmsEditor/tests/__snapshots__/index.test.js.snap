exports[`<CmsEditor /> should render with default props 1`] = `
<div>
  <div>
    <input
      onChange={[Function bound handleInputChange]}
      placeholder="Title"
      type="titleInput"
      value="" />
  </div>
  <MegadraftEditor
    editorState={
      EditorState {
        "_immutable": Object {
          "allowUndo": true,
          "currentContent": Object {
            "blockMap": Object {
              "cont2": Object {
                "characterList": Array [],
                "data": Object {},
                "depth": 0,
                "key": "cont2",
                "text": "",
                "type": "unstyled"
              }
            },
            "selectionAfter": Object {
              "anchorKey": "cont2",
              "anchorOffset": 0,
              "focusKey": "cont2",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false
            },
            "selectionBefore": Object {
              "anchorKey": "cont2",
              "anchorOffset": 0,
              "focusKey": "cont2",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false
            }
          },
          "decorator": CompositeDraftDecorator {
            "_decorators": Array [
              Object {
                "component": [Function Link],
                "strategy": [Function findLinkEntities]
              }
            ]
          },
          "directionMap": Object {
            "cont2": "LTR"
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Array [],
          "selection": Object {
            "anchorKey": "cont2",
            "anchorOffset": 0,
            "focusKey": "cont2",
            "focusOffset": 0,
            "hasFocus": false,
            "isBackward": false
          },
          "treeMap": Object {
            "cont2": Array [
              Object {
                "decoratorKey": null,
                "end": 0,
                "leaves": Array [
                  Object {
                    "end": 0,
                    "start": 0
                  }
                ],
                "start": 0
              }
            ]
          },
          "undoStack": Array []
        }
      }
    }
    onChange={[Function bound onChange]}
    placeholder="Tell a story ..." />
  <undefined
    canSubmit={false}
    onSave={[Function bound publish]}
    onSetStatus={[Function bound setStatus]}
    onToggleSpotlight={[Function bound setSpotlightToggle]}
    spotlighted={false}
    status={0} />
</div>
`;
